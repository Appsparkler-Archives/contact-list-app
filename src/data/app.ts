import { IContactFormData } from "types";

export type TAppFC = React.FC<{
  contacts: IContactFormData[];
}>;
