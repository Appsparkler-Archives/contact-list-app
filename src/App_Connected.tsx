import { App } from "./App";
import { IAppProps } from "types";
import { useCallback } from "react";
import { contactsActions } from "features/contacts/contactsSlice";
import { useAppDispatch, useAppSelector } from "app/hooks";

export const ConnectedApp: React.FC<{}> = () => {
  const contacts = useAppSelector((state) => state.contacts);

  const dispatch = useAppDispatch();

  const handleCreate: IAppProps["onCreate"] = useCallback(
    (contacts) => {
      dispatch(contactsActions.create(contacts));
    },
    [dispatch]
  );

  const handleEdit: IAppProps["onEdit"] = useCallback(
    (contact2Edit) => {
      dispatch(contactsActions.edit(contact2Edit));
    },
    [dispatch]
  );

  const handleDelete: IAppProps["onDelete"] = useCallback(
    (contactIdToDelete) => {
      dispatch(contactsActions.delete(contactIdToDelete));
    },
    [dispatch]
  );

  return (
    <App
      contacts={contacts}
      onCreate={handleCreate}
      onEdit={handleEdit}
      onDelete={handleDelete}
    />
  );
};
