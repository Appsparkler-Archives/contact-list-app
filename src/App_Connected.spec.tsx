import { render } from "@testing-library/react";
import { ConnectedApp } from "App_Connected";
import { store } from "app/store";
import { Provider } from "react-redux";

const AppRenderer = () => (
  <Provider store={store}>
    <ConnectedApp />;
  </Provider>
);

describe("App - Snapshot Tests", () => {
  it("should render App component", () => {
    const { container } = render(<AppRenderer />);
    expect(container).toMatchSnapshot();
  });
});
