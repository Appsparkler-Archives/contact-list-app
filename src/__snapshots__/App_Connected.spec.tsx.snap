// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - Snapshot Tests should render App component 1`] = `
<div>
  <div
    class="MuiBox-root css-ejw0ia"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Contacts
    </h4>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
    >
      You have 
      3
       contacts.
    </h6>
    <div
      class="MuiBox-root css-hlt1gb"
    >
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-aulpef-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          1 female
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-aulpef-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          2 male
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-aulpef-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          2 business
        </span>
      </div>
      <div
        class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-aulpef-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          1 personal
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonAddAlt1Icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 8c0-2.21-1.79-4-4-4S5 5.79 5 8s1.79 4 4 4 4-1.79 4-4zm2 2v2h3v3h2v-3h3v-2h-3V7h-2v3h-3zM1 18v2h16v-2c0-2.66-5.33-4-8-4s-8 1.34-8 4z"
            />
          </svg>
        </span>
        Add New Contact
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="7cb1248c-983a-4207-a59f-40bfa1fae564"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="7cb1248c-983a-4207-a59f-40bfa1fae564"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              class="MuiBox-root css-j7qwjs"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Ali Connors
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
              >
                SurveyHeart
              </h6>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="7cb1248c-983a-4207-a59f-40bfa1fae564"
                class="MuiAccordion-region"
                id="7cb1248c-983a-4207-a59f-40bfa1fae564"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiBox-root css-9jay18"
                  >
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="read"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ChromeReaderModeIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M13 12h7v1.5h-7zm0-2.5h7V11h-7zm0 5h7V16h-7zM21 4H3c-1.1 0-2 .9-2 2v13c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15h-9V6h9v13z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <button
                      aria-label="edit"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-1gws2xf-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="delete"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorWarning MuiIconButton-sizeMedium css-1moznky-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DeleteIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="b4fc7a3e-3522-4292-b05d-5edbdffcf3ac"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="b4fc7a3e-3522-4292-b05d-5edbdffcf3ac"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              class="MuiBox-root css-j7qwjs"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Jenny Allen
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
              >
                Societe Generale
              </h6>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="b4fc7a3e-3522-4292-b05d-5edbdffcf3ac"
                class="MuiAccordion-region"
                id="b4fc7a3e-3522-4292-b05d-5edbdffcf3ac"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiBox-root css-9jay18"
                  >
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="read"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ChromeReaderModeIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M13 12h7v1.5h-7zm0-2.5h7V11h-7zm0 5h7V16h-7zM21 4H3c-1.1 0-2 .9-2 2v13c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15h-9V6h9v13z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <button
                      aria-label="edit"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-1gws2xf-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="delete"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorWarning MuiIconButton-sizeMedium css-1moznky-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DeleteIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="e90ae941-f295-4adc-bf8a-b1d7127a60e6"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="e90ae941-f295-4adc-bf8a-b1d7127a60e6"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              class="MuiBox-root css-j7qwjs"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                Steven Covey
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="e90ae941-f295-4adc-bf8a-b1d7127a60e6"
                class="MuiAccordion-region"
                id="e90ae941-f295-4adc-bf8a-b1d7127a60e6"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiBox-root css-9jay18"
                  >
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="read"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ChromeReaderModeIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M13 12h7v1.5h-7zm0-2.5h7V11h-7zm0 5h7V16h-7zM21 4H3c-1.1 0-2 .9-2 2v13c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15h-9V6h9v13z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <button
                      aria-label="edit"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-1gws2xf-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="delete"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorWarning MuiIconButton-sizeMedium css-1moznky-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DeleteIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  ;
</div>
`;
